src/main/java/org/roaringbitmap/longlong/Roaring64NavigableMap.java

```diff
@@ -0,0 +1,1351 @@
+        // binary-search
+        return position + 1;
+      } else {
+        // This high has no bitmap: it could be between 2 highs with bitmaps
+        int insertionPosition = -position - 1;
+        return insertionPosition;
+      }
+    } else {
+      // For each deprecated buckets
+      SortedMap<Integer, BitmapDataProvider> tailMap =
+          highToBitmap.tailMap(firstHighNotValid, true);
+
+      // TODO .size on tailMap make an iterator: arg
+      int indexOk = highToBitmap.size() - tailMap.size();
+
+      Iterator<Entry<Integer, BitmapDataProvider>> it = tailMap.entrySet().iterator();
+      while (it.hasNext()) {
+        Entry<Integer, BitmapDataProvider> e = it.next();
+        int currentHigh = e.getKey();
+
+        if (compare(currentHigh, high) > 0) {
+          // No need to compute more than needed
+          break;
+        } else if (e.getValue().isEmpty()) {
+          // highToBitmap can not be modified as we iterate over it
+          if (latestAddedHigh != null && latestAddedHigh.getKey().intValue() == currentHigh) {
+            // Dismiss the cached bitmap as it is removed from the NavigableMap
+            latestAddedHigh = null;
+          }
+          it.remove();
+        } else {
+          ensureOne(e, currentHigh, indexOk);
+
+          // We have added one valid cardinality
+          indexOk++;
+        }
+
+      }
+
+      if (highToBitmap.isEmpty() || indexOk == highToBitmap.values().size()) {
+        // We have compute all cardinalities
+        allValid = true;
+      }
+
+      return indexOk;
+    }
+  }
+
+  private int binarySearch(int[] array, int key) {
+    if (signedLongs) {
+      return Arrays.binarySearch(array, key);
+    } else {
+      return unsignedBinarySearch(array, 0, array.length, key,
+          RoaringIntPacking.unsignedComparator());
+    }
+  }
+
+  private int binarySearch(int[] array, int from, int to, int key) {
+    if (signedLongs) {
+      return Arrays.binarySearch(array, from, to, key);
+    } else {
+      return unsignedBinarySearch(array, from, to, key, RoaringIntPacking.unsignedComparator());
+    }
+  }
+
+  // From Arrays.binarySearch (Comparator). Check with org.roaringbitmap.Util.unsignedBinarySearch
+  private static int unsignedBinarySearch(int[] a, int fromIndex, int toIndex, int key,
+      Comparator<? super Integer> c) {
+    int low = fromIndex;
+    int high = toIndex - 1;
+
+    while (low <= high) {
+      int mid = (low + high) >>> 1;
+      int midVal = a[mid];
+      int cmp = c.compare(midVal, key);
+      if (cmp < 0) {
+        low = mid + 1;
+      } else if (cmp > 0) {
+        high = mid - 1;
+      } else {
+        return mid; // key found
+      }
+    }
+    return -(low + 1); // key not found.
+  }
+
+  private void ensureOne(Map.Entry<Integer, BitmapDataProvider> e, int currentHigh, int indexOk) {
+
+    // sortedHighs are valid only up to some index
+    assert indexOk <= sortedHighs.length;
```

Comment:
Could you add a message to the assert, with at least `indexOf` and `sortedHighs.length` ? If it came to fail, this would be helpful more than just knowing the line.

Reply:
Improved

Reply:
You say improved but I don't see any change ...

Reply:
The fix was waiting on another machine and my remote access seemed broken. I felt it should not prevent merging the PR. Anyway, I have finally been able to push the related fix.